### PR TITLE
pkg: Support active_time_intervals in AlertmanagerConfig CRD

### LINKED ISSUE #4656: Add support for active time intervals in AlertmanagerConfig CRD
**What is missing?**

Alertmanager v0.24 (to be released soon) supports active time intervals (see https://github.com/prometheus/alertmanager/pull/2779). The functionality would need to be added to the AlertmanagerConfig CRD too.

Similar to what happens in Alertmanager, the existing `spec.muteTimeIntervals` field needs to be deprecated in favor of `spec.timeIntervals`. The operator could seamlessly merge both fields in the generated configuration:
* using `time_intervals` if the Alertmanager's `spec.version` is >= v0.24.
* `mute_time_intervals` otherwise.

**Why do we need it?**

Feature parity with upstream Alertmanager.

**Environment**

* Prometheus Operator version: N/A

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by chtcvl**
Fixes #4656

## Description

Add support for using `active_time_intervals` in AlertmanagerConfig CRD Route configuration for Alertmanager >= 0.24.

## Type of change

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

```release-note
Add support for using `active_time_intervals` in AlertmanagerConfig CRD Route configuration.
```

### CODE REVIEW

File: pkg/alertmanager/amcfg.go [hunk 1]
```diff
@@ -88,6 +88,10 @@ func checkAlertmanagerConfigRootRoute(rootRoute *route) error {
 		return errors.New("'mute_time_intervals' not permitted on root route")
 	}
 
+	if len(rootRoute.ActiveTimeIntervals) > 0 {
+		return errors.New("'active_time_intervals' not permitted on root route")
```
Comment (simonpasquier) [inline pkg/alertmanager/amcfg.go:92]: good catch!

File: pkg/apis/monitoring/v1beta1/conversion_to.go [hunk 2]
```diff
@@ -31,13 +31,14 @@ func convertRouteTo(in *Route) (*v1alpha1.Route, error) {
...
-		Matchers:          convertMatchersTo(in.Matchers),
-		MuteTimeIntervals: in.MuteTimeIntervals,
+		Receiver:            in.Receiver,
+		GroupBy:             in.GroupBy,
+		GroupWait:           in.GroupWait,
+		GroupInterval:       in.GroupInterval,
+		RepeatInterval:      in.RepeatInterval,
+		Matchers:            convertMatchersTo(in.Matchers),
+		MuteTimeIntervals:   in.MuteTimeIntervals,
+		ActiveTimeIntervals: in.ActiveTimeIntervals,
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1beta1/conversion_to.go:41]: Dont' we need a similar change in conversion_from.go?

Comment (chtcvl) [reply pkg/apis/monitoring/v1beta1/conversion_to.go:41]: Thanks, updated `conversion_from.go` accordingly

**Comment by simonpasquier** [comment]
I've restarted the failed job which is unrelated to your change :)

**Review by simonpasquier — APPROVED** [review]

**Comment by simonpasquier** [comment]
thanks!

**Pull request merged**